### ISSUE TITLE
Bug: (no-floating-promises rule doesn't check return type of .catch)

### ISSUE THREAD

**Issue by IgorWiecz** [eslint/eslint#17305]
### Environment

Node version: 18.15.3
npm version: 9.6.7
ESLint version: 8.36.0
Operating System: win32 10.0.19045

Returning a **Promise** in a **.catch** is not detected by the **no-floating-promises** rule.
Eg: _dispatch(getStatus()).catch((error: string) => Promise.reject(error))_

the type of getStatus: **AsyncThunk<string, void, AsyncThunkConfig>**

### What parser are you using?

@typescript-eslint/parser

### What did you do?

<details>
<summary>Configuration</summary>

```
<!-- Paste your configuration here -->
```
</details>

```js
  const dispatch = useAppDispatch()

  useEffect(() => {
    dispatch(getStatus()).catch((error: string) => Promise.reject(error))
  }, [dispatch])
```

### What did you expect to happen?

no-floating-promises error to show up

### What actually happened?

nothing

### Link to Minimal Reproducible Example

NA

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by Rec0iL99** [comment]
Hi @IgorWiecz, the [`no-floating-promises`](https://typescript-eslint.io/rules/no-floating-promises) is a rule exported from [`typescript-eslint`](https://github.com/typescript-eslint/typescript-eslint) and the project is not maintained by the ESLint team. The best course of action here would be to open the issue there.

Thanks for using ESLint.

**Comment by IgorWiecz** [comment]
> Hi @IgorWiecz, the [`no-floating-promises`](https://typescript-eslint.io/rules/no-floating-promises) is a rule exported from [`typescript-eslint`](https://github.com/typescript-eslint/typescript-eslint) and the project is not maintained by the ESLint team. The best course of action here would be to open the issue there.
> 
> Thanks for using ESLint.

Ok thanks:

https://github.com/typescript-eslint/typescript-eslint/issues/7135
